src/components/retrieve_from_weaviate/src/main.py

```diff
@@ -0,0 +1,55 @@
+import dask
+import pandas as pd
+from fondant.component import PandasTransformComponent
+from llama_index import ServiceContext, VectorStoreIndex
+from llama_index.embeddings import HuggingFaceEmbedding
+from llama_index.vector_stores import WeaviateVectorStore
+
+import weaviate
+
+dask.config.set({"dataframe.convert-string": False})
+
+
+class RetrieveChunks(PandasTransformComponent):
+    def __init__(
+        self,
+        *_,
+        weaviate_url: str,
+        class_name: str,
+        text_property_name: str,
+        hf_embed_model: str,
+        top_k: int
+    ) -> None:
+        """
+        Args:
+            weaviate_url: An argument passed to the component
+        """
+        # Initialize your component here based on the arguments
+        self.client = weaviate.Client(weaviate_url)
+        self.class_name = class_name
+        self.text_property_name = text_property_name
+        self.model = HuggingFaceEmbedding(hf_embed_model)
```

Comment:
This needs to align with the embeddings used in the Indexing pipeline. I think we might have to insert an embedding component in between the `load_csv` and `query_weaviate` components. That way, we can just run it with the same parameters as we do the one in the indexing pipeline.

Reply:
done. The retrieve components awaits for a "embedding" column based on which it will retrieve the chunks:)